### PR TITLE
[RI2] Converting docs from markdown to rst

### PR DESCRIPTION
**Pull request by gkunz**
This PR implements cleanup of the documentation and conversion from markdown to rst.

Implements #2649

### CODE REVIEW

**Comment by tomkivlin** [comment]
Just as an FYI as I don't want to undo any good work already done, but internally I have moved from using recommonmark as the markdown parser to myst, which allows native sphinx directives and roles within a markdown document, but the rest of the doc remains in markdown.

e.g. for images: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-images

(it also deals with copying images and other artefacts automatically, so no need for any "direct_copy" script any more.

**Comment by CsatariGergely** [comment]
> Regarding the branch, I agree that it is helpful to keep things clean this way. However, I "noticed" that Sphinx happily works with both formats in parallel - as for instance demonstrated by this PR which creates rst files among markdown files. So, we could actually convert chapters one-by-one in the same tree - thereby also minimizing the problem of diverging content between master and the integration branch. On the other hand, we'd need a commitment by everyone to go ahead with rst to avoid mixed formats. Yet, even that wouldn't be a problem, i.e., leave it up to the WSLs to decide the format they prefer.

Now I agree that this approach without the integration branch is better, so let's follow this approach. I will do the same with my pr-s.

Yes, in this way we can leave it to the sub-projects to decide on what tooling they need. On the other side I would gradually stop using recommonmark as it is not supported and forces us to use an old version of Sphinx.

**Comment by CsatariGergely** [comment]
> Just as an FYI as I don't want to undo any good work already done, but internally I have moved from using recommonmark as the markdown parser to myst, which allows native sphinx directives and roles within a markdown document, but the rest of the doc remains in markdown.
> 
> e.g. for images: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-images
> 
> (it also deals with copying images and other artefacts automatically, so no need for any "direct_copy" script any more.

Looks like a good option, but for the sub-projects handed over to GSMA rst is the desired format. For other sub-projects this could be a viable option if GitHub is able to render it correctly.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+Structure of the document
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The document first consolidates the requirements that need to be addressed by this RI in `Chapter 2 <./chapter02.md>`__. `Chapter 3 <./chapter03.md>`__ then details the requirements that need to be met in order for anyone to be able to install the RI, whether that be into a community lab, their own corporate or personal lab or some other environment such as CNF Testbed or a public cloud provider. An operational runbook is documented in `Chapter 4 <./chapter04.md>`__ to aid the deployment of the RI into a chosen environment and includes step-by-step instructions for a selection of installers as well as steps on how to validate the deployment. Finally, `Chapter 5 <./chapter05.md>`__ is a placeholder to allow the documentation of any gaps found during the development of this document and the associated artifacts.
+
+Terminology
+~~~~~~~~~~~
+
+Terminology in this document will follow `CNTT Terminology <../../../common/glossary.md>`__.
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:27]:
```suggestion
Terminology in this document will follow `Anuket Terminology <../../../common/glossary.md>`__.
```

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+-  **CNCF - CNF Conformance, Kubernetes, etc.**
+
+   -  CNCF hosts a wide variety of projects providing software components for RI2, such as Kuberntes itself, deployment tooling (e.g. kubeadm and kubespray) as well as test suites (e.g., Kubernetes conformance).
+   -  The Telco User Group (TUG) is a forum within CNCF to identify and discuss telco requirements on the Kubernetes ecosystem.
+   -  The CNF Conformance project in CNCF has a stated aim to provide "visibility into how well Cloud native Network Functions (CNFs) and the underlying Telecom platform follows cloud native principles".
+   -  The CNF Testbed is an initiative providing a framework for building and deploying technology show cases with a strong focus on telco platform requirements.
+
+The following figure gives an overview of the relationship of the communities involved.
+
+|Relationship of communities|
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:71]: What about the Figure caption?

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst [hunk 3]
```diff
@@ -0,0 +1,139 @@
+Overview
+========
+
+Introduction
+------------
+
+About the Kubernetes Reference Implementation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This document includes the requirements and deployment details of the Kubernetes-based Reference Implementation (RI2). This RI will conform to the architecture specification as defined in the `Kubernetes Based Reference Architecture (RA2) <../../../ref_arch/kubernetes/README.md>`__, including alignment with the principles and requirements of the `Reference Model <../../../ref_model/README.md>`__.
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:10]: Can we define these references at the end like the figures? Easy to change if a reference link changes or there is an error, etc. Thanks.

**Comment by collivier** [comment]
Agree @pgoyal01 . please add linkcheck (+docutils) which will automatically detect all false links to md.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst (hunk 3 above)
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter01.rst:10]: yes. rst typically makes use of named references, i.e., there is a global name defined at the target and that is being referred to from everywhere. However, we should introduce this over time as it requires the target anchors to exist.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst (hunk 2 above)
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter01.rst]: added in the latest update. I am also using the rst reference mechanism to reference the figure (by name).

**Comment by gkunz** [comment]
@collivier added doc8. No link checking since I don't think we should use absolute links.

**Comment by gkunz** [comment]
Now that the other chapters are getting merged as rst, it would be good to get this reviewed and merged as well. Thank you!

File: doc/ref_impl/cntt-ri2/README.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
+Reference Implementation 2
+==========================
+
+This is the Kubernetes-based Reference Implementation (RI-2)
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/README.rst:4]:
```suggestion
This is the Kubernetes-based Reference Implementation (RI-2).
```

File: doc/ref_impl/cntt-ri2/README.rst [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+----------------------
+
+============== ============= ===================
+Bundle.Version Date          Note
+============== ============= ===================
+0.0            15th May 2020 First Initial Draft
+4.0-alpha      25th Sep 2020 Baraque Release
+6.0-alpha      1st July 2021 Kali Release
+============== ============= ===================
+
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/README.rst:25]: do we need to a Lakelse release info

Comment (gkunz) [reply doc/ref_impl/cntt-ri2/README.rst:25]: yes, that's the problem with keeping these things open for a long time. I'll merge the latest updates. Thanks for the catching this.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+   -  The Telco User Group (TUG) is a forum within CNCF to identify and discuss telco requirements on the Kubernetes ecosystem.
+   -  The CNF Conformance project in CNCF has a stated aim to provide "visibility into how well Cloud native Network Functions (CNFs) and the underlying Telecom platform follows cloud native principles".
+   -  The CNF Testbed is an initiative providing a framework for building and deploying technology show cases with a strong focus on telco platform requirements.
+
+Figure :ref:`ri2_figure_relationship-of-communities` gives an overview of the relationship of the communities involved.
+
+.. _ri2_figure_relationship-of-communities:
+
+.. figure:: ../figures/ri2-ch01-relationship_of_communities.png
+
```
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:74]: what about the figure caption?

File: doc/ref_impl/cntt-ri2/README.rst (hunk 5 above)
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/README.rst:26]: included in https://github.com/cntt-n/CNTT/pull/2656/commits/e781dee7fd59ba28d36593c292b504b9b2e9d529

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by gkunz** [comment]
just realized that this is broken - DO NOT MERGE

**Comment by pgoyal01** [comment]
@gkunz I added a label "WIP" and recycled the approvals -- would need re-approvals from @rihabbanday and me.

**Comment by gkunz** [comment]
Fixed, sorry for the hick-up.

File: doc/ref_impl/cntt-ri2/chapters/chapter04.rst [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+
+The entire installation is divided into two stages - Host provisioning and Kubernetes provisioning. Host provisioning is the operation of preparing a host before the software stack can be installed on them. This includes (and not limited to) installing an operating system, configuring network so that the hosts are reachable via SSH, configuring storage, etc. This stage can be skipped when using preprovisioned hardware, infrastructure providers, etc. The Kubernetes provisioning stage is agnostic to the host provisioning stage, in that there is no dependency between the installer used for the Kubernetes provisioning stage and any tools used in the host provisioning stage.
+
+Prerequisites
+-------------
+
+You need one physical server acting as a jump server along with minimum of two additional servers on which RI-2 will be deployed. Please refer to `Chapter 3 <./chapter03.md>`__ for detailed information on the server and network specifications.
```
Comment (collivier) [inline doc/ref_impl/cntt-ri2/chapters/chapter04.rst:14]: this won't work anymore. You could simply run linkcheck to detect all of them.

**Review by collivier — CHANGES_REQUESTED** [review]

File: doc/ref_impl/cntt-ri2/chapters/chapter04.rst (hunk 7 above)
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter04.rst]: @collivier: fixed. the last commit switched to references where possible and passed linkcheck.

**Comment by gkunz** [comment]
@collivier I addressed your last comment. Could you please re-review?

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by gkunz** [comment]
HI @collivier: I addressed your comments. Can you please re-review and/or approve? Thanks

**Review by michaelspedersen — APPROVED** [review]
Went over the changes, focusing on the content (text, references). I did not find any immediate errors compared to the previous content.

I have not tested the build process and looked at the final output.

**Comment by collivier** [comment]
@gkunz I'm fine de merge it. I don't see why I'm not allowed to write comments  in it.
I would suggest to wrap lines to 80 via pandoc, rst -> rst (it will also mange very long url).
I think sphinx config could be simplified.

I see minor edit changes in the PR. Would it make sens to fix any typo in the latest stable branch too?

Thank you to add the linkcheck part.

**Review by collivier — APPROVED** [review]

**Pull request merged**